src/wp-admin/includes/menu.php

```diff
@@ -368,7 +368,17 @@ function sort_menu( $a, $b ) {
 }
 unset( $last_menu_key );
 
-if ( ! user_can_access_admin_page() ) {
+if ( ! admin_page_exists() ) {
+
+	/**
+	 * Fires when access to an admin page which does not exist.
+	 *
+	 * @since 4.9.0
+	 */
+	do_action( 'admin_page_not_found' );
+
+	wp_die( __( 'The requested page does not exist.' ), 403 );
```

Comment:
I imagine we want a 404 code here instead of 403.

```suggestion
	wp_die( __( 'The requested page does not exist.' ), 404 );
```